src/app/context/SettingsContext.tsx

```diff
@@ -55,19 +60,41 @@ export const SettingsProvider = ({
       });
   }, []);
 
-  // update rate
-  useEffect(() => {
-    api.getCurrencyRate().then((response) => {
-      setCurrencyRate(response.rate);
-    });
-  }, [settings.currency]);
-
-  const getFiatValue = (amount: number | string) =>
-    getFiatValueFunc({
-      amount,
-      rate: currencyRate,
-      currency: settings.currency,
-    });
+  const getCurrencyRate = async (): Promise<number> => {
+    // ensure to get the correct rate for current currency in state
+    if (settings.currency !== currencyRate.current?.currency) {
+      const response = await api.getCurrencyRate(); // gets rate from browser.storage or API
+
+      // update local ref
+      currencyRate.current = {
+        rate: response.rate,
+        currency: settings.currency,
+      };
+    }
+
+    return currencyRate.current.rate;
+  };
+
+  const getFiatValue = async (amount: number | string) => {
+    try {
+      const rate = await getCurrencyRate();
+      const value = getFiatValueUtil({
+        amount,
+        rate,
+        currency: settings.currency,
+      });
+
+      return value;
+    } catch (e) {
+      console.error(e);
+      if (e instanceof Error)
+        toast.error(
```

Comment:
will this lead to an overload of toasts if something fails?  for each payment one toast? 


Also is that message helpful for the user? the user does not know what the `SettingsProvider` and what `getFiatValue` is. 
so isn't `console.error` enough?

Reply:
Removed the toasts